provider.go

```diff
@@ -169,6 +150,48 @@ func (p *VaultProvider) Retrieve() (credentials.Value, error) {
 	return value, nil
 }
 
+// RetrieveWithoutSessionToken returns credentials that are either the master credentials or
+// a session created with AssumeRole. This allows for usecases where a token created with AssumeRole
+// wouldn't work.
```

Comment:
Should this conditional be lifted up into the caller? It feels dangerous having a conditional based on some global state which defaults to exposing master credentials if something goes wrong looking for the role. Two clearly named functions might be nicer;

```go
if role, ok := p.profiles[p.profile]["role_arn"]; ok {
  credentials, err := GetAssumedRoleCredentials(role)
} else {
  credentials, err := GetMasterCredentials()
}
```

Reply:
Unfortunately `Retrieve` is the interface method here.